src/subdomains/supporting/payment/services/transaction-request.service.ts

```diff
@@ -85,6 +123,21 @@ export class TransactionRequestService {
     }
   }
 
+  getPaymentType(fiatPaymentMethod: PaymentMethod): PaymentType {
+    switch (fiatPaymentMethod) {
+      case FiatPaymentMethod.BANK:
+        return PaymentType.SEPA_CREDIT;
+      case FiatPaymentMethod.INSTANT:
+        return PaymentType.SEPA_INSTANT_CREDIT;
+      case FiatPaymentMethod.CARD:
+        return PaymentType.CREDIT_CARD;
+      case CryptoPaymentMethod.CRYPTO:
+        return PaymentType.CRYPTO_CURRENCY;
+      default:
+        throw new Error(`No payment type for fiat payment ${fiatPaymentMethod}`);
+    }
+  }
```

Comment:
Easier with a map:
```ts
const PaymentMethodMap: { [method in PaymentMethod]: PaymentType } = {
  [FiatPaymentMethod.BANK]: PaymentType.SEPA_CREDIT,
  [FiatPaymentMethod.INSTANT]: PaymentType.SEPA_INSTANT_CREDIT,
  [FiatPaymentMethod.CARD]: PaymentType.CREDIT_CARD,
  [CryptoPaymentMethod.CRYPTO]: PaymentType.CRYPTO_CURRENCY,
};
```
and this will generate a build error, if a PaymentMethod is not mapped.